htlcswitch/interceptable_switch.go

```diff
@@ -596,15 +634,16 @@ func (f *interceptedForward) Packet() InterceptedPacket {
 			ChanID: f.packet.incomingChanID,
 			HtlcID: f.packet.incomingHTLCID,
 		},
-		OutgoingChanID: f.packet.outgoingChanID,
-		Hash:           f.htlc.PaymentHash,
-		OutgoingExpiry: f.htlc.Expiry,
-		OutgoingAmount: f.htlc.Amount,
-		IncomingAmount: f.packet.incomingAmount,
-		IncomingExpiry: f.packet.incomingTimeout,
-		CustomRecords:  f.packet.customRecords,
-		OnionBlob:      f.htlc.OnionBlob,
-		AutoFailHeight: f.autoFailHeight,
+		OutgoingChanID:    f.packet.outgoingChanID,
+		Hash:              f.htlc.PaymentHash,
+		OutgoingExpiry:    f.htlc.Expiry,
+		OutgoingAmount:    f.htlc.Amount,
+		IncomingAmount:    f.packet.incomingAmount,
+		IncomingExpiry:    f.packet.incomingTimeout,
+		CustomRecords:     f.packet.customRecords,
+		OnionBlob:         f.htlc.OnionBlob,
+		AutoFailHeight:    f.autoFailHeight,
+		CustomPeerRecords: f.packet.wireRecords,
```

Comment:
So we already have the incoming htlc here (`htlc`), therefore we can directly access the `htlc.ExtraData` field.

Reply:
Yeap a field of the desired type exists there, but this actually doesn't contain any `ExtraData` as the struct is re-constructed by the link , as seen [here](https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/link.go#L3371-L3376)